packages/apollo-gateway/src/__tests__/integration/provides.feature

```diff
@@ -0,0 +1,59 @@
+Feature: provides directive
+
+  Scenario: does not have to go to another service when field is given
+    Given federated accounts, books, inventory, product, reviews services
+    When I execute
+      """
+      query GetReviewers {
+        topReviews {
+          author {
+            username
+          }
+        }
+      }
+      """
+    Then the reviews service should be called
+    And the accounts service should not be called
+    And the response should be
+      """
+      {
+        "data": {
+          "topReviews": [
+            { "author": { "username": "@ada" } },
+            { "author": { "username": "@ada" } },
+            { "author": { "username": "@complete" } },
+            { "author": { "username": "@complete" } },
+            { "author": { "username": "@complete" } }
+          ]
+        }
+      }
+      """
+
+  Scenario: does not load fields provided even when going to other service
+    Given federated accounts, books, inventory, product, reviews services
+    When I execute
+      """
+        query GetReviewers {
+          topReviews {
+            author {
+              username
+              name
+            }
+          }
+        }
+      """
+    Then the accounts, reviews services should be called
```

Comment:
There's not a simple way that I can think of to ensure a specific resolver isn't called. In the existing test, there's a mocked resolver and a check to ensure that it isn't called. I could get into that, but I'll hold off until later if we decide we really want that. I figure the query plan would _probably_ be sufficient

Reply:
@trevor-scheer Do you know if checking the query plan would be sufficient in this case? I'm inclined to agree with @JakeDawkins here.

Reply:
Yep this can be verified with the query plan alone!